drivers/usb/device/usb_dc_stm32.c

```diff
@@ -757,7 +744,7 @@ int usb_dc_ep_check_cap(const struct usb_dc_ep_cfg_data * const cfg)
 		return -1;
 	}
 
-	if (ep_idx > (USB_NUM_BIDIR_ENDPOINTS - 1)) {
+	if (ep_idx > (DT_USB_NUM_BIDIR_ENDPOINTS - 1)) {
```

Comment:
Nit: use same form as L194 for consistency
```suggestion
	if (ep_idx >= DT_USB_NUM_BIDIR_ENDPOINTS) {
```